src/Components/AdvancedSearch/AdvancedSearchModal.tsx

```diff
@@ -43,11 +42,35 @@ const AdvancedSearchModal: React.FC<IAdvancedSearchProps> = (props) => {
     });
     const { t } = useTranslation();
     const titleId = useId('advanced-search-modal-title');
+    const filteredTwins = useRef<IADTTwin[]>([]);
     const additionalProperties = useRef(new Set<string>());
-    // TODO: Make this actually call query
+    const searchForTwinAdapterData = useAdapter({
+        adapterMethod: (params: AdapterMethodParamsForAdvancedSearchADTwins) =>
+            adapter.advancedSearchADTTwins(params),
+        refetchDependencies: [],
+        isAdapterCalledOnMount: false
+    });
+
     const executeQuery = (query: string) => {
-        alert(query);
+        searchForTwinAdapterData.callAdapter({
+            query: query,
+            // TODO: Define onScroll method on list to change continuation token
+            continuationToken: null
+        });
     };
+
+    useEffect(() => {
+        const selectedTwins: IADTTwin[] = [];
+        if (searchForTwinAdapterData.adapterResult?.result?.data) {
+            searchForTwinAdapterData.adapterResult.result.data.value.map(
+                (twinData: IADTTwin) => {
+                    selectedTwins.push(twinData);
+                }
+            );
+        }
+        filteredTwins.current = selectedTwins;
```

Comment:
why store the results in a ref? Don't you want to trigger a render when it updates that list?

Reply:
It will trigger a render, since the results are passed in as a prop to the list.

Reply:
Hmmm. but why woudl this parent rerender to do the passing? I think this should be stored in state instead. That seems like an unnessecary risk. We _want_ this component to render when this chagnes so I don't see why we would make it a ref.

Reply:
Why do you want the parent to re-render? There are no visual updates in the parent that require a re-render.